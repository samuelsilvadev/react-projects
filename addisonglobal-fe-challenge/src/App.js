import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './Home';

import Header from './components/header/Header';

import './App.scss';

const App = () => (
	<>
		<Header />
		<Switch>
			<Route exact path="/" component={Home} />
		</Switch>
	</>
);

export default App;
