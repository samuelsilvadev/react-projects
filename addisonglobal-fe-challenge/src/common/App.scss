* {
	box-sizing: border-box;
	margin: 0;
	padding: 0;
}

body {
	margin: 0;
	padding: 0;
	font-family: sans-serif;
}

.main {
	padding: 0 15px;
}
