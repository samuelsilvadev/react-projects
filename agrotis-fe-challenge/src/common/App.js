import React from 'react';
import styled from 'styled-components';
import { Route, Switch } from 'react-router-dom';

import { Header, HEADER_HEIGHT } from './components';
import Home from './pages/Home';
import { GlobalStyle } from './Global.style';
import { gteSmallMedia } from './media.style';

const Main = styled.main`
	background-color: #ecf0f1;
	min-height: calc(100vh - ${HEADER_HEIGHT});
	padding: 2rem;

	${gteSmallMedia} {
		padding: 2rem 4rem;
	}
`;

const App = () => (
	<>
		<GlobalStyle />
		<Header />
		<Main>
			<Switch>
				<Route exact path="/" component={Home} />
			</Switch>
		</Main>
	</>
);

export default App;
