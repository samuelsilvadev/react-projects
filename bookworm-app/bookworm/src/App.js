import React from 'react';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import HomePage from './components/pages/HomePage'
import LoginPage from './components/pages/LoginPage';
import DashBoardPage from './components/pages/DashBoardPage';
import UserRoute from './components/routes/UserRoute';
import GuestRoute from './components/routes/GuestRoute';
import SignUp from './components/pages/SignUpPage';

const App = ({ location }) => (
  <div className="ui container">
    <Route location={location} path="/" exact component={HomePage} />
    <GuestRoute location={location} path="/login" component={LoginPage} />
    <GuestRoute location={location} path="/signup" component={SignUp} />
    <UserRoute location={location} path="/dashboard" component={DashBoardPage} />
  </div>
);

App.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired
};

export default App;
