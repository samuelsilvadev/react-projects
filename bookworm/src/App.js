import React from 'react';

const App = () => (
  <div>Hi</div>
);

export default App;
