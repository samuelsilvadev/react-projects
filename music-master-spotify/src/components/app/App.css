:root {
    --black: #000;
    --white: #fff;
    --font-size-base: 16px; 
}

body {
    background-color: var(--black);
    color: var(--white); 
    font-size: var(--font-size-base);
}

.App {
    text-align: center;
    padding: 5%;
}