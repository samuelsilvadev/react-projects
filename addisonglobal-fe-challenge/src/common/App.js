import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Home } from './pages';

import Header from './components/header/Header';
import BetSlip from './components/betslip';

import { useAsideContext, withAsideContext } from './contexts/AsideContext';

import './App.scss';

const App = () => {
	const { handleOpen, handleClose, isAsideVisible } = useAsideContext();

	return (
		<>
			<Header onAsideOpen={handleOpen} />
			<main className="main">
				<Switch>
					<Route exact path="/" component={Home} />
				</Switch>
				<BetSlip open={isAsideVisible} onClose={handleClose} />
			</main>
		</>
	);
};

export default withAsideContext(App);
