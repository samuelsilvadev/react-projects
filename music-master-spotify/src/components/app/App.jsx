import React from 'react';
import { FormGroup, FormControl, InputGroup, Glyphicon } from 'react-bootstrap';

import './App.css';
import Profile from '../profile/Profile';
import Gallery from '../gallery/Gallery';

class App extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            query: '',
            artist: null,
        };
        this._changeValueField = this._changeValueField.bind(this);
        this._search = this._search.bind(this);
        this._handleEnterKeyPessed = this._handleEnterKeyPessed.bind(this);
        this._convertJsonToCorretObjectOnState = this._convertJsonToCorretObjectOnState.bind(this);
    }

    _changeValueField(e) {
        this.setState({ query: e.target.value });
    }

    _search() {
        const BASE_URL = 'https://api.spotify.com/v1/search?';
        const FETCH_URL = `${BASE_URL}q=${this.state.query}&type=artist&limit=1`;
        window.fetch(FETCH_URL, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': process.env.REACT_APP_SPOTIFY_TOKEN,
            }
        })
        .then(this._handleResponseSearch)
        .then(this._convertJsonToCorretObjectOnState)
        .catch(this._handleErrorResponseSearch);
    }

    _handleEnterKeyPessed(e) {
        if (e.key === 'Enter') {
            this._search();
        }
    }

    _convertJsonToCorretObjectOnState(bigJSON) {
        const [artist] = bigJSON.artists.items;
        this.setState({
            artist
        });
    }

    _handleResponseSearch(resp) {
        return resp.json();
    }

    _handleErrorResponseSearch(error) {
        console.error(error);
    }

    render() {
        return (
            <div className="App">
                <h1 className="App__title">Music Master</h1>
                <FormGroup>
                    <InputGroup>
                        <FormControl
                            type="text"
                            placeholder="Search for an Artist"
                            query={this.state.query}
                            onChange={this._changeValueField}
                            onKeyPress={this._handleEnterKeyPessed}
                        />
                        <InputGroup.Addon onClick={this._search}>
                            <Glyphicon glyph="search"></Glyphicon>
                        </InputGroup.Addon>
                    </InputGroup>
                </FormGroup>
                <Profile artist={this.state.artist} />
                <Gallery />
            </div>
        );
    }
}

export default App;