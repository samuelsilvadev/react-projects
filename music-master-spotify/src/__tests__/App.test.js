import React from 'react';
import { shallow } from 'enzyme';

import App from '../components/app/App';

describe('<App />', () => {
    it('should render correctly component', () => {
        const tree = shallow(<App />);
        
        expect(tree).toMatchSnapshot();
    });
});