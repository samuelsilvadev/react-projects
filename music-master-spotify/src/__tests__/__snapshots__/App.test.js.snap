// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="App__title"
>
          music master
</h1>,
        <FormGroup
          bsClass="form-group"
          className="App__search-form"
          role="search"
>
          <InputGroup
                    bsClass="input-group"
          >
                    <FormControl
                              aria-label="search for an artist"
                              aria-required="true"
                              bsClass="form-control"
                              componentClass="input"
                              onChange={[Function]}
                              onKeyPress={[Function]}
                              placeholder="search for an artist"
                              query=""
                              required={true}
                              role="textbox"
                              type="text"
                    />
                    <InputGroupAddon
                              bsClass="input-group-addon"
                              onClick={[Function]}
                              role="button"
                    >
                              <Glyphicon
                                        bsClass="glyphicon"
                                        glyph="search"
                              />
                    </InputGroupAddon>
          </InputGroup>
</FormGroup>,
        null,
        <Gallery />,
      ],
      "className": "App",
      "role": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "music master",
          "className": "App__title",
        },
        "ref": null,
        "rendered": "music master",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <InputGroup
            bsClass="input-group"
>
            <FormControl
                        aria-label="search for an artist"
                        aria-required="true"
                        bsClass="form-control"
                        componentClass="input"
                        onChange={[Function]}
                        onKeyPress={[Function]}
                        placeholder="search for an artist"
                        query=""
                        required={true}
                        role="textbox"
                        type="text"
            />
            <InputGroupAddon
                        bsClass="input-group-addon"
                        onClick={[Function]}
                        role="button"
            >
                        <Glyphicon
                                    bsClass="glyphicon"
                                    glyph="search"
                        />
            </InputGroupAddon>
</InputGroup>,
          "className": "App__search-form",
          "role": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "input-group",
            "children": Array [
              <FormControl
                aria-label="search for an artist"
                aria-required="true"
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                onKeyPress={[Function]}
                placeholder="search for an artist"
                query=""
                required={true}
                role="textbox"
                type="text"
/>,
              <InputGroupAddon
                bsClass="input-group-addon"
                onClick={[Function]}
                role="button"
>
                <Glyphicon
                                bsClass="glyphicon"
                                glyph="search"
                />
</InputGroupAddon>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "search for an artist",
                "aria-required": "true",
                "bsClass": "form-control",
                "componentClass": "input",
                "onChange": [Function],
                "onKeyPress": [Function],
                "placeholder": "search for an artist",
                "query": "",
                "required": true,
                "role": "textbox",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "input-group-addon",
                "children": <Glyphicon
                  bsClass="glyphicon"
                  glyph="search"
/>,
                "onClick": [Function],
                "role": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "glyphicon",
                  "glyph": "search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="App__title"
>
            music master
</h1>,
          <FormGroup
            bsClass="form-group"
            className="App__search-form"
            role="search"
>
            <InputGroup
                        bsClass="input-group"
            >
                        <FormControl
                                    aria-label="search for an artist"
                                    aria-required="true"
                                    bsClass="form-control"
                                    componentClass="input"
                                    onChange={[Function]}
                                    onKeyPress={[Function]}
                                    placeholder="search for an artist"
                                    query=""
                                    required={true}
                                    role="textbox"
                                    type="text"
                        />
                        <InputGroupAddon
                                    bsClass="input-group-addon"
                                    onClick={[Function]}
                                    role="button"
                        >
                                    <Glyphicon
                                                bsClass="glyphicon"
                                                glyph="search"
                                    />
                        </InputGroupAddon>
            </InputGroup>
</FormGroup>,
          null,
          <Gallery />,
        ],
        "className": "App",
        "role": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "music master",
            "className": "App__title",
          },
          "ref": null,
          "rendered": "music master",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <InputGroup
              bsClass="input-group"
>
              <FormControl
                            aria-label="search for an artist"
                            aria-required="true"
                            bsClass="form-control"
                            componentClass="input"
                            onChange={[Function]}
                            onKeyPress={[Function]}
                            placeholder="search for an artist"
                            query=""
                            required={true}
                            role="textbox"
                            type="text"
              />
              <InputGroupAddon
                            bsClass="input-group-addon"
                            onClick={[Function]}
                            role="button"
              >
                            <Glyphicon
                                          bsClass="glyphicon"
                                          glyph="search"
                            />
              </InputGroupAddon>
</InputGroup>,
            "className": "App__search-form",
            "role": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "input-group",
              "children": Array [
                <FormControl
                  aria-label="search for an artist"
                  aria-required="true"
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  placeholder="search for an artist"
                  query=""
                  required={true}
                  role="textbox"
                  type="text"
/>,
                <InputGroupAddon
                  bsClass="input-group-addon"
                  onClick={[Function]}
                  role="button"
>
                  <Glyphicon
                                    bsClass="glyphicon"
                                    glyph="search"
                  />
</InputGroupAddon>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "aria-label": "search for an artist",
                  "aria-required": "true",
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "onChange": [Function],
                  "onKeyPress": [Function],
                  "placeholder": "search for an artist",
                  "query": "",
                  "required": true,
                  "role": "textbox",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "input-group-addon",
                  "children": <Glyphicon
                    bsClass="glyphicon"
                    glyph="search"
/>,
                  "onClick": [Function],
                  "role": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "glyphicon",
                    "glyph": "search",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
